sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -684,6 +685,24 @@ object TypeCoercion {
     }
   }
 
+  /**
+   * The DIV operator always returns long-type value .
+   * This rule cast the integral inputs to long type, to avoid overflow during calculation.
+   */
+  object IntegralDivision extends TypeCoercionRule {
```

Comment:
Plz add fine-grained tests  for this rule in `TypeCoercionSuite`.

Reply:
Added tests in `TypeCoercionSuite`